versions/3.0.4.md

```diff
@@ -133,10 +133,22 @@ In order to preserve the ability to round-trip between YAML and JSON formats, YA
 
 ### <a name="documentStructure"></a>OpenAPI Description Structure
 
-An OpenAPI Description MAY be made up of a single document or be divided into multiple, connected parts at the discretion of the author. In the latter case, `$ref` fields MUST be used in the specification to reference those parts as follows from the [JSON Schema](https://json-schema.org) definitions.  In a multi-document description, the document containing the [OpenAPI Object](#oasObject) is known as the **entry OpenAPI document.**
+An OpenAPI Description (OAD) MAY be made up of a single document or be divided into multiple, connected parts at the discretion of the author. In the latter case, [Reference Object](#referenceObject) and [Path Item Object](#pathItemObject) `$ref` keywords, as well as the [Link Object](#linkObject) `operationRef` keyword, are used.  In a multi-document description, the document containing the [OpenAPI Object](#oasObject) is known as the **entry OpenAPI document.**
 
 It is RECOMMENDED that the entry OpenAPI document be named: `openapi.json` or `openapi.yaml`.
 
+#### <a name="structuralInteroperability"></a>Structural Interoperability
+
+When parsing an OAD, JSON or YAML objects are parsed into specific Objects (such as [Operation Objects](#operationObject), [Response Objects](#responseObject), [Reference Objects](#referenceObject), etc.) based on the parsing context.  Depending on how references are arranged, a given JSON or YAML object can be parsed in multiple different contexts:
```

Comment:
should "are parsed" be changed to "MUST BE parsed" as per [BCP 14](https://tools.ietf.org/html/bcp14) / [RFC2119](https://tools.ietf.org/html/rfc2119) / [RFC8174](https://tools.ietf.org/html/rfc8174)?

Reply:
In this case I'd say no, it's not necessary, as the whole spec defines that sort of parsing.  This is a place where I'm trying to describe what tools tend to do without becoming prescriptive about how they do it.  Or getting too deep into the details of various options.

For example, some tools parse the underlying format and the OpenAPI aspects more-or-less at one (at least I assume somoene out there uses and event-based streaming parser or something of that sort).  Others do two separate passes.  Some do both passes every time they hit the reference, others do the JSON/YAML pass once and re-do the OAS part each time they hit the reference (particularly in 2.0).

So I didn't want someone to read my phrasing here and decide it mandated a specific implementation.  But I'm open to other perspectives.